frontend/src/components/BarChartPreview.tsx

```diff
@@ -15,12 +15,19 @@ type Props = {
   title: string;
   summary: string;
   bars: Array<string>;
-  data?: Array<object>;
+  data?: Array<any>;
   summaryBelow: boolean;
 };
 
 const BarChartPreview = (props: Props) => {
   const colors = useColors(props.bars.length);
+  const yAxisType = useCallback(() => {
+    return props.data &&
+      props.data.every((row) => typeof row[props.bars[0]] === "number")
+      ? "number"
+      : "category";
+  }, [props]);
+
```

Comment:
Soon we need to start moving some of these logic to a `DatasetParsing.ts` file or something like that, so that we don't repeat this logic in every Chart component.